### ISSUE TITLE
EPIC: migrate SAVE-cloud processing to KAFKA

### ISSUE THREAD

**Issue by akuleshov7** [saveourtool/save-cloud#1046]

**Comment by icemachined** [comment]
One of the problems of kafka resolves,
then you call rest endpoint sequentially, you can have race condition inside code with the changes which can be applied in different order. kafka guarantees ordering, Probably rest also have some mechanisms to cope with it but they can be quite complicated such as throttling or shelving changes out of order.

**Comment by nulls** [comment]
Suggestion is to start with communication between `save-agent` and `save-orchestrator\save-sandbox` -- `save-agent` sends heartbeats to orchestrator and receives a task to do by response. It can be replaced to Kafka without affecting the rest components

**Comment by petertrr** [comment]
Another feature that can be implemented using a message broker is managing the lifecycle and monitoring monitoring the state of an execution (see #331, #764, #886 as different attempts on this). Event of execution creation can be fired and then consumed by the next actor (e.g. by orchestrator after backend stores a new execution into the DB). In case an actor is incapable of performing the task (internal error), the event hangs in the queue until either the service retries the task or a monitoring service (backend, probably) notices it as a case for global timeout.

Basically, the same can be done now with constant polling of `execution` table, but Kafka is hopefully a more suitable solution.

Or, all participating services can fire execution status updates periodically into a dedicated topic, and too long silence there would mean that processing is abnormally finished.
